src/app/core/auth/server-auth-request.service.ts

```diff
@@ -27,8 +30,14 @@ export class ServerAuthRequestService extends AuthRequestService {
    * @param href The href to send the request to
    * @protected
    */
-  protected createShortLivedTokenRequest(href: string): GetRequest {
-    return Object.assign(new GetRequest(this.requestService.generateRequestId(), href), {
+  protected createShortLivedTokenRequest(href: string): PostRequest {
+    let options = new HttpHeaders();
+    options = options.set('Content-Type', 'application/json; charset=utf-8');
+    options = options.set(XSRF_REQUEST_HEADER, this.tokenExtractor.getToken());
+    let requestOptions = {
+      headers: options,
+    };
+    return Object.assign(new PostRequest(this.requestService.generateRequestId(), href, {}, requestOptions), {
```

Comment:
@ConfusionOrb221 : Maybe I'm confused, but it seems like if we switch this to a `POST` request we should be able to just use the exact same code a `BrowserAuthRequestService` (which already does a POST).  It's odd to me that this method needs so much more than the one line here: https://github.com/DSpace/dspace-angular/blob/main/src/app/core/auth/browser-auth-request.service.ts#L31

For instance, it doesn't seem like we should need to manually add the `XSRF_REQUEST_HEADER` here as it should be added automatically by the `xsrf.interceptor.ts` for any `POST` request.  That's the job of the `xsrf.interceptor.ts` already... it automatically adds the XSRF token to all requests that need it. See https://github.com/DSpace/dspace-angular/blob/main/src/app/core/xsrf/xsrf.interceptor.ts#L78-L86

Did you somehow find that the `xsrf.interceptor.ts` isn't working properly in this scenario?  I'm just confused by the new code added here & how this is solving the issues with https://github.com/DSpace/DSpace/pull/8628

Reply:
This code was updated in the latest commits by Art. Essentially `Xsrf.interceptor.ts` can't get the xsrf token because when running in prod the function is ran on the node server. Art implemented into this function the ability to grab xsrf tokens first from the rest side then attach that to the request aswell here.